### PR TITLE
Add `*-pattern` custom message formatting

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Ref: https://github.com/stylelint/stylelint-config-standard/issues/263

> Is there anything in the PR that needs further explanation?

This PR updates the following rules:

- `custom-media-pattern`
- `custom-property-pattern`
- `keyframes-name-pattern`
- `selector-class-pattern`
- `selector-id-pattern`
- `selector-nested-pattern`

Also, this PR makes the rules' message more unified and simpler like this (see also <https://github.com/stylelint/stylelint/pull/6391#discussion_r986307057> for details):

```diff
- Expected custom media query name "${name}" to match pattern "${pattern}"
+ Expected "${name}" to match pattern "${pattern}"
```

### CODE REVIEW

File: .changeset/many-tools-build.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": minor
+---
+
+Added: custom message formatting to `custom-media-pattern`, `custom-property-pattern`, `keyframes-name-pattern`, `selector-class-pattern`, and `selector-id-pattern`
```
Comment (ybiquitous) [inline .changeset/many-tools-build.md:5]: [ask] Too long?

Comment (jeddy3) [reply .changeset/many-tools-build.md:5]:
```suggestion
Added: `*-pattern` custom message formatting
```
To be consistent with the likes of "Added: `font-weight-notation` autofix'".

Comment (ybiquitous) [reply .changeset/many-tools-build.md]: Nice! Thanks.

File: lib/rules/custom-media-pattern/index.js [hunk 2]
```diff
@@ -9,7 +9,8 @@ const { isRegExp, isString } = require('../../utils/validateTypes');
 const ruleName = 'custom-media-pattern';
 
 const messages = ruleMessages(ruleName, {
-	expected: (pattern) => `Expected custom media query name to match pattern "${pattern}"`,
+	expected: (name, pattern) =>
+		`Expected custom media query name "${name}" to match pattern "${pattern}"`,
```
Comment (jeddy3) [inline lib/rules/custom-media-pattern/index.js:13]:
```suggestion
		`Expected "${name}" to match pattern "${pattern}"`,
```

We now generally favour not repeating the rule name in the message if we quote the actual thing.

e.g.

`Expected "${name}" to match pattern "${pattern}" (custom-media-pattern)`

Did we want to make these changes in this pull request or a follow-up one?

**Review by jeddy3 — COMMENTED** [review]
LGTM. I've asked one question.

File: lib/rules/custom-media-pattern/index.js (hunk 2 above)
Comment (ybiquitous) [reply lib/rules/custom-media-pattern/index.js:13]: That's nice and much simpler!

We now have the following rules with a `... to match pattern` message:

```console
$ git grep 'to match pattern'
lib/rules/comment-pattern/index.js:11:    expected: (pattern) => `Expected comment to match pattern "${pattern}"`,
lib/rules/custom-media-pattern/index.js:13:        `Expected custom media query name "${name}" to match pattern "${pattern}"`,
lib/rules/custom-property-pattern/index.js:17:        `Expected custom property name "${name}" to match pattern "${pattern}"`,
lib/rules/keyframes-name-pattern/index.js:14:        `Expected keyframe name "${keyframeName}" to match pattern "${pattern}"`,
lib/rules/selector-class-pattern/index.js:17:        `Expected class selector "${selector}" to match pattern "${pattern}"`,
lib/rules/selector-id-pattern/index.js:14:        `Expected ID selector "${selector}" to match pattern "${pattern}"`,
lib/rules/selector-nested-pattern/index.js:13:        `Expected nested selector "${selector}" to match pattern "${pattern}"`,
```

In this pull request, I'll add custom message formatting and apply a new message pattern to all other rules than `comment-pattern`.

Comment (ybiquitous) [reply lib/rules/custom-media-pattern/index.js]: Fixed via 19f313e3f

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

**Comment by ybiquitous** [comment]
Thank you so much for your excellent review. 😊

**Pull request merged**